substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -1032,18 +1090,33 @@ impl<T: Config> Pallet<T> {
 	/// wrong.
 	pub fn do_add_validator(who: &T::AccountId, prefs: ValidatorPrefs) {
 		if !Validators::<T>::contains_key(who) {
-			// maybe update sorted list.
-			let _ = T::VoterList::on_insert(who.clone(), Self::weight_of(who))
-				.defensive_unwrap_or_default();
+			Validators::<T>::insert(who, prefs);
+			T::EventListeners::on_validator_add(who);
+		} else {
+			Validators::<T>::insert(who, prefs);
 		}
-		Validators::<T>::insert(who, prefs);
 
 		debug_assert_eq!(
 			Nominators::<T>::count() + Validators::<T>::count(),
-			T::VoterList::count()
+			T::VoterList::iter()
+				.filter(|v| Self::status(&v) != Ok(StakerStatus::Idle))
```

Comment:
Wondering if this assertion would ever fail? If a voter does not exist in Nominator or Validator map, it will be returned as `Idle` in the `Self::status()`. 
Update: All good, this actually would fail if account is not bonded. Though if am not wrong this count might be same as bonded map count? Unfortunately, it is not a CountedStorageMap yet.